src/io/flutter/dart/FlutterDartAnalysisServer.java

```diff
@@ -110,5 +183,21 @@ private void processNotification(JsonObject response) {
         }
       }
     }
+    return true;
+  }
+
+  /**
+   * Must use it right before sending any offsets and lengths to the AnalysisServer.
+   */
+  private int getOriginalOffset(@Nullable final VirtualFile file, final int convertedOffset) {
+    final Method method = ReflectionUtil.getDeclaredMethod(analysisService.getClass(), "getOriginalOffset", VirtualFile.class, int.class);
```

Comment:
Is this the API we want on the analysis server class, and if so, can you:
- file a PR to increase the access
- add a TODO: here about when we can remove the reflection

Reply:
I opened https://github.com/JetBrains/intellij-plugins/pull/572 to make the method public and added TODO.